File: VB/NonPersistentObjectsDemo.Module/Module.vb
Imports System
Imports DevExpress.ExpressApp
Imports System.ComponentModel
Imports DevExpress.ExpressApp.DC
Imports System.Collections.Generic
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.ExpressApp.Updating
Imports DevExpress.ExpressApp.Xpo
Imports NonPersistentObjectsDemo.Module.BusinessObjects

Namespace NonPersistentObjectsDemo.Module

    ' For more typical usage scenarios, be sure to check out https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.ModuleBase.
    Public NotInheritable Partial Class NonPersistentObjectsDemoModule
        Inherits ModuleBase

        Private nonPersistentObjectSpaceHelper As NonPersistentObjectSpaceHelper

        Public Sub New()
            InitializeComponent()
            BaseObject.OidInitializationMode = OidInitializationMode.AfterConstruction
        End Sub

        Public Overrides Function GetModuleUpdaters(ByVal objectSpace As IObjectSpace, ByVal versionFromDB As Version) As IEnumerable(Of ModuleUpdater)
            Dim updater As ModuleUpdater = New DatabaseUpdate.Updater(objectSpace, versionFromDB)
            Return New ModuleUpdater() {updater}
        End Function

        Public Overrides Sub Setup(ByVal application As XafApplication)
            MyBase.Setup(application)
            ' Manage various aspects of the application UI and behavior at the module level.
            AddHandler application.SetupComplete, AddressOf Application_SetupComplete
        End Sub

        Private Sub Application_SetupComplete(ByVal sender As Object, ByVal e As EventArgs)
            nonPersistentObjectSpaceHelper = New NonPersistentObjectSpaceHelper(CType(sender, XafApplication), GetType(BaseObject))
            nonPersistentObjectSpaceHelper.AdapterCreators.Add(Sub(npos)
                Dim types = New Type() {GetType(Account), GetType(Message)}
                Dim map = New ObjectMap(npos, types)
                Dim tmp_TransientNonPersistentObjectAdapter = New TransientNonPersistentObjectAdapter(npos, map, New PostOfficeFactory(map))
            End Sub)
        End Sub

        Public Overrides Sub CustomizeTypesInfo(ByVal typesInfo As ITypesInfo)
            MyBase.CustomizeTypesInfo(typesInfo)
            CalculatedPersistentAliasHelper.CustomizeTypesInfo(typesInfo)
        End Sub
    End Class
End Namespace

File: CS/NonPersistentObjectsDemo/NonPersistentObjectsDemo.Module/Module.cs
using System.ComponentModel;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.DC;
using DevExpress.Persistent.Base;
using DevExpress.ExpressApp.Model;
using DevExpress.ExpressApp.Actions;
using DevExpress.ExpressApp.Editors;
using DevExpress.ExpressApp.Updating;
using DevExpress.ExpressApp.Model.Core;
using DevExpress.ExpressApp.Model.DomainLogics;
using DevExpress.ExpressApp.Model.NodeGenerators;
using DevExpress.Persistent.BaseImpl.EF;
using NonPersistentObjectsDemo.Module.BusinessObjects;
using NonPersistentObjectsDemo.Module.ServiceClasses;

namespace NonPersistentObjectsDemo.Module;

// For more typical usage scenarios, be sure to check out https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.ModuleBase.
public sealed class NonPersistentObjectsDemoModule : ModuleBase {
    private NonPersistentObjectSpaceHelper nonPersistentObjectSpaceHelper;
    public NonPersistentObjectsDemoModule() {
		// 
		// NonPersistentObjectsDemoModule
		// 
		RequiredModuleTypes.Add(typeof(DevExpress.ExpressApp.SystemModule.SystemModule));
		RequiredModuleTypes.Add(typeof(DevExpress.ExpressApp.Objects.BusinessClassLibraryCustomizationModule));
		RequiredModuleTypes.Add(typeof(DevExpress.ExpressApp.ConditionalAppearance.ConditionalAppearanceModule));
      
    }
    public override IEnumerable<ModuleUpdater> GetModuleUpdaters(IObjectSpace objectSpace, Version versionFromDB) {
        ModuleUpdater updater = new DatabaseUpdate.Updater(objectSpace, versionFromDB);
        return new ModuleUpdater[] { updater };
    }
    public override void Setup(XafApplication application) {
        base.Setup(application);
        // Manage various aspects of the application UI and behavior at the module level.
        application.SetupComplete += Application_SetupComplete;
    }
    private void Application_SetupComplete(object sender, EventArgs e) {
        nonPersistentObjectSpaceHelper = new NonPersistentObjectSpaceHelper((XafApplication)sender, typeof(BaseObject));
        nonPersistentObjectSpaceHelper.AdapterCreators.Add(npos => {
            var types = new Type[] { typeof(Account) };
            var map = new ObjectMap(npos, types);
            PostOfficeFactory factory =(PostOfficeFactory)((XafApplication)sender).ServiceProvider.GetService(typeof(PostOfficeFactory));

            new TransientNonPersistentObjectAdapter(npos, map, factory);
        });
    }
}
